### PR TITLE
json encoding and decoding

### PR DESCRIPTION
**Pull request by havardpe**
@arnej27959 please review
@mpolden FYI

### CODE REVIEW

File: client/go/internal/vespa/slime/json.go [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+}
+
+func (e *jsonEncoder) fmt(format string, stuff ...any) {
+	if e.err == nil {
+		_, e.err = fmt.Fprintf(e.output, format, stuff...)
+	}
+}
+
+func (e *jsonEncoder) put(value byte) {
+	e.fmt(string(value))
```
Comment (arnej27959) [inline client/go/internal/vespa/slime/json.go:88]: write to e.output directly

File: client/go/internal/vespa/slime/inserter.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	Insert(value Value) Value
+}
+
+type myInserter func(value Value) Value
+
+func (f myInserter) Insert(value Value) Value {
+	return f(value)
+}
+
+func InsertRoot(root *Value) myInserter {
```
Comment (arnej27959) [inline client/go/internal/vespa/slime/inserter.go:15]: consider declaring this to return Inserter

**Review by arnej27959 — COMMENTED** [review]
minor fix needed

**Comment by havardpe** [comment]
@arnej27959 PTAL

**Review by arnej27959 — DISMISSED** [review]
👍

**Review by arnej27959 — APPROVED** [review]
👍

**Pull request merged**